hamocc/mo_dmsph.F90

```diff
@@ -0,0 +1,146 @@
+! Copyright (C) 2020  J. Tjiputra
+!
+! This file is part of BLOM/iHAMOCC.
+!
+! BLOM is free software: you can redistribute it and/or modify it under the
+! terms of the GNU Lesser General Public License as published by the Free
+! Software Foundation, either version 3 of the License, or (at your option)
+! any later version.
+!
+! BLOM is distributed in the hope that it will be useful, but WITHOUT ANY
+! WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+! FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for
+! more details.
+!
+! You should have received a copy of the GNU Lesser General Public License
+! along with BLOM. If not, see https://www.gnu.org/licenses/.
+
+module mo_dmsph
+
+  implicit none
+  private
+  public :: get_pi_ph,get_dmsph,pi_ph_path,with_dmsph
+
+  ! Activate/deactivate calculation of DMS as a function of pH.
+  logical :: with_dmsph = .false.
+
+  ! Path to input data, set through namelist in hamocc_init.F
+  character(len=256),save    :: pi_ph_path = ''
+
+  ! Length of surface PI pH record from file
+  ! Assume monthly record by default (otherwise change get_dmsph)
+  integer, parameter :: pi_ph_record = 12
+
+  ! dms_ph scaling factor
+  real, parameter :: dms_gamma = 0.87
+
+  ! surface PI pH field
+  real, dimension(:,:,:), allocatable :: pi_ph
+
+CONTAINS
+
+
+  SUBROUTINE GET_PI_PH(kpie,kpje,omask)
+  !**********************************************************************
+
+    use mo_control_bgc, only: io_stdo_bgc
+    use netcdf,         only: nf90_noerr,nf90_nowrite,nf90_close,nf90_open
+    use mod_xc,         only: mnproc,xchalt
+
+    implicit none
+    INTEGER, INTENT(in) :: kpie,kpje
+    INTEGER ::i,j,l
+
+    REAL,intent(in) ::omask(kpie,kpje)
+
+    ! define the fields
+    REAL :: pi_ph_in(kpie,kpje,pi_ph_record)
+    INTEGER ncid,ncstat
+
+    ! allocate pi_ph field
+    if(.not. allocated(pi_ph)) call alloc_pi_ph(kpie,kpje)
+
+    !
+    ! Open netCDF data file
+    !
+    IF(mnproc==1) THEN
+       ncstat = NF90_OPEN(trim(pi_ph_path)//'MONTHLY_PI_PH.nc',        &
+     &                   NF90_NOWRITE, ncid)
+       write(io_stdo_bgc,*) 'HAMOCC: opening MONTHLY_PI_PH file'
```

Comment:
Would be good to not have a fixed file name. The filename should at least indicate the grid, probably also the model version (or run) plus start/end dates from which the climatology is made